File: app/src/main/java/com/thebook/bottomnav/ui/home/SimpleViewModel.java
package com.thebook.bottomnav.ui.home;

/**
 * SimpleViewModel - Data model class for individual movie items
 * 
 * This class represents a single movie object with the essential information
 * needed to display movie cards in RecyclerView lists. It acts as a data 
 * transfer object between the JSON data source and the UI components.
 * 
 * Note: Despite the name "ViewModel", this is actually a data model/POJO class,
 * not an Android Architecture Component ViewModel.
 */
public class SimpleViewModel {
    // The poster image resource name (e.g., "tt37", "tt38")
    private String poster;
    
    // The movie title (e.g., "Avatar", "Pirates of the Caribbean")
    private String title;
    
    // The drawable resource ID for the movie poster image
    private int image;

    /**
     * Default constructor required for object instantiation
     */
    public SimpleViewModel() {
        // Default constructor
    }

    /**
     * Parameterized constructor for creating a movie object with all data
     * @param title The movie title
     * @param poster The poster resource name
     * @param image The drawable resource ID for the poster image
     */
    public SimpleViewModel(String title, String poster, int image) {
        this.title = title;
        this.poster = poster;
        this.image = image;
    }

    // Getter and setter methods for accessing private fields

    /**
     * @return The drawable resource ID for the movie poster image
     */
    public int getImage() {
        return image;
    }

    /**
     * @param image The drawable resource ID to set for the movie poster
     */
    public void setImage(int image) {
        this.image = image;
    }

    /**
     * @return The poster resource name (used to find drawable resources)
     */
    public String getPoster() {
        return poster;
    }

    /**
     * @param poster The poster resource name to set
     */
    public void setPoster(String poster) {
        this.poster = poster;
    }

    /**
     * @return The movie title
     */
    public String getTitle() {
        return title;
    }

    /**
     * @param title The movie title to set
     */
    public void setTitle(String title) {
        this.title = title;
    }
}

File: app/src/main/java/com/thebook/bottomnav/ui/home/SimpleViewModel.kt
package com.thebook.bottomnav.ui.home

/**
 * ========================================
 * MOVIE DATA MODEL
 * ========================================
 * 
 * PURPOSE:
 * This represents a single movie in our app. It's a simple data container
 * that holds all the information we need to display a movie card.
 * 
 * WHAT IT CONTAINS:
 * - title: The movie name (e.g., "Avatar", "Spectre")
 * - posterUrl: Web URL where the movie poster image is located
 * - resourceId: Local Android resource ID (for offline images)
 * 
 * WHY IT'S A DATA CLASS:
 * Kotlin data classes automatically provide:
 * - equals() and hashCode() methods
 * - toString() method for debugging
 * - copy() method for creating modified copies
 * - Component functions for destructuring
 * 
 * USAGE EXAMPLE:
 * val movie = Movie("Avatar", "https://example.com/avatar.jpg")
 * println(movie.title) // Prints: Avatar
 */
data class Movie(
    val title: String = "",           // Movie title displayed to user
    val posterUrl: String = "",       // URL for downloading poster image
    val resourceId: Int = 0           // Local resource ID (fallback/offline)
) {
    /**
     * CONVENIENCE CONSTRUCTOR
     * For backwards compatibility - creates movie with just title and local resource
     */
    constructor(title: String, resourceId: Int) : this(title, "", resourceId)
    
    /**
     * UTILITY METHODS
     */
    
    /**
     * Checks if this movie has a valid poster URL for downloading
     */
    fun hasOnlinePoster(): Boolean = posterUrl.isNotEmpty()
    
    /**
     * Checks if this movie has a local resource image
     */
    fun hasLocalResource(): Boolean = resourceId != 0
    
    /**
     * Gets display-friendly string for debugging
     */
    override fun toString(): String = "Movie(title='$title', hasOnline=${hasOnlinePoster()}, hasLocal=${hasLocalResource()})"
}

/**
 * MIGRATION HELPER
 * Type alias to maintain compatibility with existing code that uses SimpleViewModel
 * TODO: Remove this after updating all references to use Movie instead
 */
typealias SimpleViewModel = Movie
